### PR TITLE
[master] Feature: salt.states.virt support on_crash, on_poweroff and on_lockfailure

### LINKED ISSUE #60699: [FEATURE REQUEST] salt.states.virt: support on_crash
**Describe the solution you'd like**
It would be nice if salt.states.virt.running and/or salt.states.virt.defined supported managing the `on_crash` setting, or possibly all of the [events settings](https://libvirt.org/formatdomain.html#events-configuration).

### PR DESCRIPTION
**Pull request by darshanip**
Fixes #60699 

### What issues does this PR fix or reference?
#60699 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/virt.py [hunk 1]
```diff
@@ -3619,6 +3658,30 @@ def update(
 
         .. versionadded:: 3003
 
+    :param stop_on_crash:
+        If set to ``True`` the guest will stop instead of rebooting.
+        This is specially useful when creating a virtual machine with an installation cdrom or
+        an autoinstallation needing a special first boot configuration.
+        Defaults to ``False``
+
+        .. versionadded:: ?
```
Comment (twangboy) [inline salt/modules/virt.py:3667]: 3007

File: salt/modules/virt.py [hunk 2]
```diff
@@ -3619,6 +3658,30 @@ def update(
...
+
+    :param stop_on_poweroff:
+        If set to ``True`` the guest will stop instead of rebooting.
+        This is specially useful when creating a virtual machine with an installation cdrom or
```
Comment (twangboy) [inline salt/modules/virt.py:3671]: especially... you get the idea

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/virt.py [hunk 3]
```diff
@@ -2211,12 +2220,36 @@ def init(
...
 
         .. versionadded:: 3003
 
+    :param stop_on_crash:
+        If set to ``True`` the guest will stop instead of rebooting.
+        This is especially useful when creating a virtual machine with an installation cdrom or
+        an auto-installation needing a special first boot configuration.
+        Defaults to ``False``
+
+        .. versionadded:: 3007
```
Comment (Ch3LL) [inline salt/modules/virt.py:2235]: Can you change all of the 3007 versions to `3007.0`

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Can you also add a changelog

**Pull request closed without merging**